app/src/main/java/org/astraea/partitioner/smoothPartitioner/SmoothWeightPartitioner.java

```diff
@@ -72,35 +72,40 @@ public int partition(
       case IN_Dependency:
```

Comment:
You will change the state from `Start_Dependency` to `IN_Dependency`, but it is not sync. 
```java
dependencyManager.transitionTo(DependencyManager.State.IN_Dependency);
```

It will be dangerous if another thread is changing the state.